django/contrib/admin/filters.py

```diff
@@ -111,19 +126,45 @@ def lookups(self, request, model_admin):
     def expected_parameters(self):
         return [self.parameter_name]
 
-    def choices(self, changelist):
+    def get_facet_counts(self, changelist):
+        original_value = self.used_parameters.get(self.parameter_name)
+        filtered_qs = changelist.get_queryset(self.request)
+        counts = {}
+        for i, choice in enumerate(self.lookup_choices):
+            self.used_parameters[self.parameter_name] = choice[0]
+            lookup_qs = self.queryset(self.request, filtered_qs)
+            if lookup_qs is not None:
+                counts[f"{i}__c"] = models.Count(
+                    changelist.pk_attname,
+                    filter=lookup_qs.query.where,
+                )
+        self.used_parameters[self.parameter_name] = original_value
+        return filtered_qs.aggregate(**counts)
+
+    def get_choices(self, changelist):
+        add_facets = changelist.add_facets
+        facet_counts = self.get_facet_counts(changelist) if add_facets else None
+        count = None
         yield {
             "selected": self.value() is None,
             "query_string": changelist.get_query_string(remove=[self.parameter_name]),
             "display": _("All"),
+            "count": sum(facet_counts.values()) if add_facets else None,
         }
-        for lookup, title in self.lookup_choices:
+        for i, (lookup, title) in enumerate(self.lookup_choices):
+            if add_facets:
+                count = facet_counts.get(f"{i}__c", -1)
+                if count != -1:
+                    title = f"{title} ({count})"
+                else:
+                    title = f"{title} (-)"
```

Comment:
Maybe walrus :smiley: Sorry for nitpicking
```suggestion
                if (count := facet_counts.get(f"{i}__c", -1)) != -1:
                    title = f"{title} ({count})"
                else:
                    title = f"{title} (-)"
```

Reply:
No worries :+1: updated